corehq/util/pagination.py

```diff
@@ -182,6 +195,31 @@ def _get_event_handler(self):
         else:
             return ResumableIteratorEventHandler(self)
 
+    def retry(self, item_id, max_retry=3):
+        """Add document to be yielded at end of iteration
+
+        Iteration order of retry documents is undefined. All retry
+        documents will be yielded after the initial non-retry phase of
+        iteration has completed, and every retry document will be
+        yielded each time the iterator is stopped and resumed during the
+        retry phase. This method is relatively inefficient because it
+        forces the iteration state to be saved to couch. If you find
```

Comment:
fwiw, and not sure if i'm speaking out of place here since i haven't been following the whole thing, but before i knew simon was working on this i was implementing the same thing with a similar approach. essentially making a ReusableResumableIterator where i just copied a bunch of the ResumableIterator code haha. anyways, one of my first changes was to move the state of the iterator to the meta database. as daniel mentioned though, i added the database name to the iterator_name to ensure that it would be unique in the meta db. while i thought keeping the iteration state in one time migrations made sense these were my reasons for saving the state in a separate db:

1. the migration state was now going to be run multiple times for reindexes (instead of just a one time thing) and something we'd probably want to have on a dashboard somewhere. creating a view for one database that got all the in-progress iterators over having a view in every database to view the status of the iterator seemed better

2. as already mentioned above, i didn't want to create a separate abstraction for iterating over SQL vs. Couch. ideally the reindexer would describe how to iterate and the iterator would iterate over it and save the state of iterator and not have to know about where that data lived